### ISSUE TITLE
Add Vim to MySQL container (and others?)

### ISSUE THREAD

**Issue by mikeerickson** [laradock/laradock#804]
I have a need for vim on my workspace container (makes editing files MUCH easier) . 

If anybody needs it elsewhere (I am also thinking of adding it to the mysql container) please let me know.

**Comment by zeroc0d3** [comment]
I don't know if it could be more space for vim + plugins inside container...
For some review, a sample autoinstaller vim + plugins which has been tested in my public [repos](https://github.com/zeroc0d3/vim-ide).

**Comment by winfried-van-loon** [comment]
Isn't Vi available within the containers? While I know that Vim is a lot friendlier to work with, compared to Vi. However, is it really necessary to install Vim in the containers: how often do you actually use Vi or Vim within a container (other than Workspace) and wouldn't it be as easy to just run `apt install vim`?

I don't think it's a bad idea - just wondering if it would be needed for containers other than Workspace.

**Comment by mikeerickson** [comment]
@winfried-van-loon No, it is not installed by default so far as I can tell.  I need to rebuild the workspace container (but not sure how to do that right now, ie which DockerFile should be used 56 or 70) .  

Further `apt-get install vim` is indeed the core requirement, just want to do that automatically.  Surely something I can do manually (along with a few other packages I always install) .  But I am configuring a container for our entire dev team (22 devs) and I dont want each dev to have to install vim manually)

In short:

```
$ apt update
$ apt install vim
```

**Comment by winfried-van-loon** [comment]
After testing with a clean Docker installation, I've got the following results.

```
E: Unable to locate package vi
```
```
vim is already the newest version (2:7.4.1689-3ubuntu1.2).
```
Looks like your request is already a reality! 👍 
Of course: if I'm missing anything here, please let me know and feel free to re-open.

**Comment by winfried-van-loon** [comment]
I... only tested the Workspace70, not the others. Could be an unfair test. More to come.

**Comment by winfried-van-loon** [comment]
To test it for workspace php56, we would need to have workspace php56 working in the first place. Related to issue #748 laradock/workspace#8 and pull request laradock/workspace#15

**Comment by winfried-van-loon** [comment]
Okay, after doing some research, I found this line:
https://github.com/laradock/workspace/blob/1.3/Dockerfile#L59

Currently, Laradock uses Workspace 1.3, which means that laradock/workspace with tag 1.3 is active, which always contains Vim. Future versions (1.4, 1.5 and the open pull requests) don't remove Vim either, so I'm not sure why your Workspace can't find Vim.

Not closing this issue as the discussion (about adding Vim to other containers and including plugins to Vim, as @zeroc0d3 suggested) is still open.

**Comment by mikeerickson** [comment]
@winfried-van-loon Actually, my issue was that it is not installed in the mysql container.

**Comment by thorerik** [comment]
the mysql container, or any other, other than workspace should have vim by default. If you find yourself in the need to have vim in a container, you're knee deep in debugging something, in which case, you can spend those 3 seconds it take to do `apt update && apt install vim`. 

The idea here is that these are images that should be at least a step closer to production images, so it even more so makes it a bad idea to add unnecessary packages got containers.

**Comment by mikeerickson** [comment]
I will take it upon myself to close the issue as I opened it.  Clearly folks like @thorerik feel I am wrong.  Yes, I can easily do it myself so I will just leave it at that.

**Comment by ryancwalsh** [comment]
I'm new to Docker and don't understand the recent point by thorerik. I've already many times been annoyed at having to manually install vim. (I agree with this suggestion by @mikeerickson .) 

I'm now trying to learn how to at least change my *own* Docker setup so that I don't need to keep doing this. 

I don't know where to put this:

```
apt update
apt install vim
```
